Gems/EMotionFX/Code/EMotionFX/Rendering/Common/RotateManipulator.cpp

```diff
@@ -404,51 +479,41 @@ namespace MCommon
             }
 
             // the intersection variables
-            AZ::Vector3 intersectA, intersectB;
+            AZ::Vector3 rotationDirection, rayHitStart, rayHitEnd;
 
             // set rotation mode to rotation around the x axis, if the following intersection conditions are fulfilled
             // innerAABB not hit, outerAABB hit, innerBoundingSphere hit and angle between cameraRollAxis and clickPosition > pi/2
-            if ((mousePosRay.Intersects(m_xAxisInnerAabb) == false ||
-                 (camera->GetProjectionMode() == MCommon::Camera::PROJMODE_ORTHOGRAPHIC && m_xAxisVisible)) &&
-                mousePosRay.Intersects(m_xAxisAabb, &intersectA, &intersectB) &&
-                mousePosRay.Intersects(m_innerBoundingSphere) &&
-                MCore::Math::ACos(camRollAxis.Dot((intersectA - m_position).GetNormalized())) > MCore::Math::halfPi)
+            if (internalCheckIntersectXAxis(
+                    camera->GetProjectionMode(), mousePosRay, camRollAxis, rotationDirection))
```

Comment:
These refactoring changes do look nicer, I just think we can improve the internal implementation to remove the duplication